src/main/java/uk/ac/ebi/embl/api/validation/submission/SubmissionValidationPlan.java

```diff
@@ -151,149 +160,185 @@ public void execute() throws ValidationEngineException {
 			if (contigDB != null)
 				contigDB.close();
 		}
+		return validationResult;
 	}
 
 	public static Set<String> getUnplacedEntryNames() {
 		return FileValidationCheck.unplacedEntryNames;
 	}
-	private void createMaster() throws ValidationEngineException
+	private ValidationResult createMaster() throws ValidationEngineException
 	{
+		ValidationResult result = new ValidationResult();
 		try
 		{
 			masterCheck = new MasterEntryValidationCheck(options);
 			if(options.processDir.isPresent()
 					&& Files.exists(Paths.get(String.format("%s%s%s",options.processDir.get(),File.separator,masterFlagFileName)))
 					&& masterCheck.getMasterEntry() != null ) {
-				return;
+				return result;
 			}
 
-			if(!masterCheck.check())
-				throw new ValidationEngineException("Master entry validation failed",ReportErrorType.VALIDATION_ERROR );
-			else if(!options.isRemote)
+			result = masterCheck.check();
+			if(!result.isValid()) {
+				result.setDefaultOrigin(new DefaultOrigin("Master entry validation failed"));
+				return result;
+			}
+			if(!options.isRemote)
 			     flagValidation(FileType.MASTER);
-		}catch(Exception e)
+		}catch(ValidationEngineException e)
 		{
 			throwValidationEngineException(FileType.MASTER,e,"master.dat");
+		} catch (IOException e) {
+			throw new ValidationEngineException(e);
 		}
+
+		return result;
 	}
 
-	private void validateChromosomeList() throws ValidationEngineException
+	private ValidationResult validateChromosomeList() throws ValidationEngineException
 	{
+		ValidationResult result = new ValidationResult();
 		if(options.processDir.isPresent()&&Files.exists(Paths.get(String.format("%s%s%s",options.processDir.get(),File.separator,chromosomelistFlagFileName))))
-         return;
-		String fileName=null;
-
-		try
-		{
+         return result;
+		String fileName = null;
+		try {
 			check = new ChromosomeListFileValidationCheck(options);
-			for(SubmissionFile chromosomeListFile:options.submissionFiles.get().getFiles(FileType.CHROMOSOME_LIST))
-			{
+			for (SubmissionFile chromosomeListFile : options.submissionFiles.get().getFiles(FileType.CHROMOSOME_LIST)) {
 				fileName= chromosomeListFile.getFile().getName();
-
-				if(!check.check(chromosomeListFile))
-					throwValidationCheckException(FileType.CHROMOSOME_LIST,chromosomeListFile);
+				result = check.check(chromosomeListFile);
+				if (!result.isValid()) {
+					result.setDefaultOrigin(new DefaultOrigin(getValidationErrorMessage(FileType.CHROMOSOME_LIST, chromosomeListFile, check)));
+					return result;
+				}
 			}
-		}catch(Exception e)
-		{
-			throwValidationEngineException(FileType.CHROMOSOME_LIST,e,fileName);
+		} catch (ValidationEngineException e) {
+			throwValidationEngineException(FileType.CHROMOSOME_LIST, e, fileName);
 		}
+		return result;
```

Comment:
Now sure why we don't just do:

catch(Exception e)
		{
			throwValidationEngineException(FileType. CHROMOSOME_LIST,e,"master.dat");
		}

Reply:
I think, there should not be any unexpected cases should occur here(cases like Null pointer exception , we should validate properly and throw ValidateEngineException).

But there is no harm in catching Exception to avoid any surprises.Will do it.